_posts/2024-11-19-release-notes-3.6.0.md

```diff
@@ -0,0 +1,207 @@
+The goal of these changes is to simplify and narrow the syntax rules required to create a given instance. To name a few:
+
+- you can now replace the `with` keyword with `:` when defining the simple type classes,
+- context bounds can now be named and aggregated using `T : {A, B}` syntax,
+- given methods defining requring contextual arguments can now be defined and chained using value
+- conditional givens can also be defined with parameters
+- by-name givens can be defined using conditional given with empty parameters list
+
+```scala
+trait Order[T]:
+  extension(values: Seq[T]) def toSorted: Seq[T] = ???
+  def compare(x: T, y: T): Int
+
+// No need for `with` keyword
+given Order[Int]:
+  def compare(x: Int, y: Int): Int = ???
+
+// Named given using named context bound parameter
+given listOrdering: [T: Order as elementOrder] => Order[List[T]]:
+  def compare(x: List[T], y: List[T]): Int = elementOrder.compare(x.head, y.head)
+
+trait Show[T]:
+  extension(value: T) def asString: String
+
+// Aggregated context parameters
+def showOrdered[T: {Order as unusedName, Show}](values: Seq[T]): Unit =
+  values.toSorted.map(_.asString).foreach(println)
+
+// Conditional givens where a contextual instance of Config is required to create an instance of Factory
+trait Config
+trait Factory
+class MemoizingFactory(config: Config) extends Factory
+given (config: Config) => Factory = MemoizingFactory(config)
+
+// By-name given
+trait Context
+given context: () => Context = ???
+```
+
+Other changes to type classes involve the stabilisation of context bounds for type members.
+The next mechanism allows to definition of an abstract given instance that needs to be provided by the class implementing trait that defines abstract given.
+
+```scala
+trait Collection:
+  // abstract member context-bound
+  type Element: Order
+
+class List[T: Order] extends Collection:
+  type Element = T
+  // given Order[Element] = deferred // generated by compiler, uses class context bound
+
+class Set[T] extends Collection:
+  type Element = T
+  override given Order[Element] = ??? // custom implementation provided by the user
+```
+
+See the updated [Contextual Abstractions](https://scala-lang.org/api/3.6.0/docs/docs/reference/contextual/givens.html) chapter of the Scala 3 reference guide to learn more about these changes.
+
+_**Note**: It is important not to confuse changes under SIP-64 with the [experimental modularity improvements](https://dotty.epfl.ch/docs/reference/experimental/typeclasses.html) available under `-language:experimental.modularity` and `-source:future`. These changes are still being developed in the experimental phase and would require SIP committee acceptance before stabilisation.
+
+## SIP-56 Amendment: Match types extractors follow aliases and singletons
```

Comment:
@sjrd Can I ask you for a technical review or some improvements, as you're the one responsible for match type specification? Probably the improvement can be better explained

Reply:
It's good.